[libpq] x64-osx-dynamic build error

**Host Environment**
- OS: macOS  12.0.1
- Compiler: Apple clang version 13.0.0 (clang-1300.0.29.3)

**To Reproduce**
Steps to reproduce the behavior:
`vcpkg install libpq:x64-osx-dynamic`

**Failure logs** 
Computing installation plan...
The following packages will be built and installed:
    libpq[bonjour,core,openssl,zlib]:x64-osx-dynamic -> 12.9#1
Detecting compiler hash for triplet x64-osx-dynamic...
Restored 0 packages from /Users/Apple/.cache/vcpkg/archives in 27.13 us. Use --debug to see more details.
Starting package 1/1: libpq:x64-osx-dynamic
Building package libpq[bonjour,core,openssl,zlib]:x64-osx-dynamic...
-- Using community triplet x64-osx-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /Users/Apple/vcpkg/triplets/community/x64-osx-dynamic.cmake
-- Using cached postgresql-12.9.tar.bz2.
-- Cleaning sources at /Users/Apple/vcpkg/buildtrees/libpq/src/tgresql-12-07bd5aa2da.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /Users/Apple/vcpkg/downloads/postgresql-12.9.tar.bz2
-- Applying patch patches/windows/install.patch
-- Applying patch patches/windows/win_bison_flex.patch
-- Applying patch patches/windows/openssl_exe_path.patch
-- Applying patch patches/windows/Solution.patch
-- Applying patch patches/windows/MSBuildProject_fix_gendef_perl.patch
-- Applying patch patches/windows/msgfmt.patch
-- Applying patch patches/windows/python_lib.patch
-- Applying patch patches/windows/fix-compile-flag-Zi.patch
-- Applying patch patches/windows/minimize_install.patch
-- Using source at /Users/Apple/vcpkg/buildtrees/libpq/src/tgresql-12-07bd5aa2da.clean
-- Getting CMake variables for x64-osx-dynamic-dbg
-- Getting CMake variables for x64-osx-dynamic-rel
-- Configuring x64-osx-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /bin/bash -c "V=1 ././configure  \"--with-openssl\" \"--with-zlib\" \"--without-readline\" \"--disable-silent-rules\" \"--verbose\" \"--enable-shared\" \"--disable-static\" \"--enable-debug\" \"--prefix=/Users/Apple/vcpkg/installed/x64-osx-dynamic/debug\" \"--bindir=\\${prefix}/../tools/libpq/debug/bin\" \"--sbindir=\\${prefix}/../tools/libpq/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/libpq\""
    Working Directory: /Users/Apple/vcpkg/buildtrees/libpq/x64-osx-dynamic-dbg
    Error code: 1
    See logs for more information:
      /Users/Apple/vcpkg/buildtrees/libpq/config-x64-osx-dynamic-dbg-out.log
      /Users/Apple/vcpkg/buildtrees/libpq/config-x64-osx-dynamic-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_make.cmake:796 (vcpkg_execute_required_process)
  ports/libpq/portfile.cmake:287 (vcpkg_configure_make)
  scripts/ports.cmake:142 (include)


Error: Building package libpq:x64-osx-dynamic failed with: BUILD_FAILED

**Additional context**
Error in the
 /Users/Apple/vcpkg/buildtrees/libpq/x64-osx-dynamic-dbg/config.log file
line 8660: dyld[72665]: Library not loaded: @rpath/libssl.1.1.dylib
[config.log](https://github.com/microsoft/vcpkg/files/7789300/config.log)
[config-x64-osx-dynamic-dbg-err.log](https://github.com/microsoft/vcpkg/files/7789322/config-x64-osx-dynamic-dbg-err.log)
[config-x64-osx-dynamic-dbg-out.log](https://github.com/microsoft/vcpkg/files/7789323/config-x64-osx-dynamic-dbg-out.log)

--- a/ports/libpq/portfile.cmake
+++ b/ports/libpq/portfile.cmake
@@ -299,8 +299,11 @@ else()
         DETERMINE_BUILD_TRIPLET
         OPTIONS
             ${BUILD_OPTS}
+        OPTIONS_RELEASE
+            "DYLD_FALLBACK_LIBRARY_PATH=${CURRENT_INSTALLED_DIR}/lib:${CURRENT_INSTALLED_DIR}/debug/lib"
         OPTIONS_DEBUG
             --enable-debug
+            "DYLD_FALLBACK_LIBRARY_PATH=${CURRENT_INSTALLED_DIR}/debug/lib:${CURRENT_INSTALLED_DIR}/lib"
     )
 
     if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")

--- a/ports/libpq/vcpkg.json
+++ b/ports/libpq/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "libpq",
   "version": "14.4",
-  "port-version": 1,
+  "port-version": 2,
   "description": "The official database access API of postgresql",
   "homepage": "https://www.postgresql.org/",
   "license": "PostgreSQL",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4058,7 +4058,7 @@
     },
     "libpq": {
       "baseline": "14.4",
-      "port-version": 1
+      "port-version": 2
     },
     "libpqxx": {
       "baseline": "7.7.3",

--- a/versions/l-/libpq.json
+++ b/versions/l-/libpq.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d452cd83bf4c62032f2ecbf66b23d75261f6ed88",
+      "version": "14.4",
+      "port-version": 2
+    },
+    {
       "git-tree": "bdf23f137f7f72e57555542fc392a8ec8e0ebd28",
       "version": "14.4",
       "port-version": 1

